src/main/java/liqp/Template.java

```diff
@@ -689,4 +714,66 @@ private void putStringKey(boolean convertValueToMap, String key, Object value, M
             map.put(key, value);
         }
     }
+
+    /**
+     * Parse the ParseTree object and create a meaningful expression from the root.
+     *
+     * @return BlockNode object that contains the parsed expression which will be used for rendering
+     *          the actual string.
+     */
+    private LNode parseRootVisit(ParseTree root, NodeVisitor nodeVisitor) {
+        BlockNode node = new BlockNode();
+
+        if (Objects.nonNull(root) && root.getChildCount() > 0) {
+            ParseTree children = root.getChild(0);
+            int childrenCount = children.getChildCount();
+
+            for (int index = 0; index < childrenCount; index++) {
+                try {
+                    LNode childLNode = nodeVisitor.visit(children.getChild(index));
+                    if (Objects.nonNull(childLNode)) {
+                        node.add(childLNode);
+                    }
+                }
+                catch (Exception exception) {
+                    logError("Exception occurred while visiting child node with nodeVisitor: ", exception.getMessage());
+                }
+            }
+        }
+        return node;
+    }
+
+    private static String processPlaceholders(String inputString, Matcher placeholderMatcher, Map<String, Object> variables) {
+        StringBuilder result = new StringBuilder();
+        int lastMatchEnd = 0;
+
+        while (Objects.nonNull(placeholderMatcher) && placeholderMatcher.find()) {
+            try {
+                result.append(inputString, lastMatchEnd, placeholderMatcher.start());
+                String placeholder = placeholderMatcher.group();
+
+                Template template = parse(placeholder);
+                String currentResultString = template.render(variables);
+
+                result.append(currentResultString);
+                lastMatchEnd = placeholderMatcher.end();
+            } catch (Exception exception) {
```

Comment:
this also Exception and workflow service also exception? this itself wouldn't leave any room right? should we make this though parserException alone ?

Reply:
yes that looks very general but I think here we can get many exceptions like:

StringIndexOutOfBoundsException,
IllegalStateException,
NullPointerException,
IllegalArgumentException


changing the catch to not append empty string because that will be same as doing nothing in String builder.